game_view.h

```diff
@@ -37,9 +38,25 @@ class game_view
   ///Gets a const ref to m_game
   const game& get_game() const noexcept {return m_game; }
 
+  /// Get next view
+  view_mode get_next_view() const
+  {
+    return m_next_view;
+  }
+
+  // Get the size of the window
+  const sf::Vector2f get_window_size() { return m_window_size; }
```

Comment:
Yup, it is useful to be able to read the window size. 
However, it can be made more const-correct and nooexcept-correct [F.6: If your function must not throw, declare it noexcept](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rf-noexcept) by doing:

```c++
const sf::Vector2f& get_window_size() const noexcept { return m_window_size; }
```

Note that we return by `const&` as we do not need a copy of this data, just the original one (see also Core Guidelines).
Adding the `const` suffices in making us happy.

Reply:
Thanks, I wasn't aware of F.6!
For const ref, I should know better 🤦

Reply:
```suggestion
  const sf::Vector2f& get_window_size() const noexcept { return m_window_size; }
```